src/List.js

```diff
@@ -47,33 +50,64 @@ const List = () => {
     setfilteredItems(filterItems);
   };
 
+  const getClassName = (nextPurchase, lastPurchased) => {
+    if (
+      lastPurchased &&
+      Date.now() - lastPurchased > 2 * nextPurchase * oneDayInMilliSecond
+    ) {
+      return 'inactive';
+    }
+    if (nextPurchase <= 7) {
+      return 'soon';
+    } else if (nextPurchase < 30) {
+      return 'kind-of-soon';
+    } else if (nextPurchase >= 30) {
+      return 'not-so-soon';
+    }
+  };
+
   return (
     <div>
       <h1>Items</h1>
-      <input type="search" value={searchTerm} onChange={handleSearch}></input>
+      <label>
+        Search:
+        <input type="search" value={searchTerm} onChange={handleSearch}></input>
+      </label>
       {items.length === 0 ? (
         <span>Your list is empty! Please add an item.</span>
       ) : filteredItems.length === 0 ? (
         <span>There's no match for {searchTerm}.</span>
       ) : (
         <ul>
           {filteredItems.map(item => (
-            <li key={item.id}>
-              <input
-                type="checkbox"
-                onChange={handlePurchase}
-                id={item.id}
-                checked={
-                  (item.lastPurchased &&
-                    now < item.lastPurchased + oneDayInMilliSecond) ||
-                  false
-                }
-                disabled={
-                  item.lastPurchased &&
-                  now < item.lastPurchased + oneDayInMilliSecond
+            <li
+              key={item.id}
+              className={getClassName(item.nextPurchase, item.lastPurchased)}
+            >
+              <label
+                aria-label={
+                  'this ' +
+                  item.name +
+                  ' will need to be purchase ' +
```

Comment:
Maybe we can conditionally render the message based on what getClassName returns. If get className returns inactive, the message will read "this {item} will need to be purchased inactive"

Reply:
> Maybe we can conditionally render the message based on what getClassName returns. If get className returns inactive, the message will read "this {item} will need to be purchased inactive"

Good thought, @MabelM1 ! Maybe stating how many days until the next purchase instead of "soon", "kind of soon," etc? Then you don't really have to do any extra work.